ConvivaTestApp/src/androidTest/java/com/bitmovin/analytics/conviva/testapp/TestBase.kt

```diff
@@ -0,0 +1,306 @@
+package com.bitmovin.analytics.conviva.testapp
+
+import android.content.Context
+import android.content.Intent
+import androidx.test.core.app.ActivityScenario
+import androidx.test.core.app.ApplicationProvider
+import com.bitmovin.analytics.conviva.ConvivaAnalytics
+import com.bitmovin.analytics.conviva.ConvivaConfiguration
+import com.bitmovin.analytics.conviva.MetadataOverrides
+import com.bitmovin.player.config.media.DASHSource
+import com.bitmovin.player.config.media.SourceItem
+import com.conviva.api.*
+import com.conviva.api.player.PlayerStateManager
+import io.mockk.*
+import org.junit.Assert
+import org.junit.Before
+import java.security.SecureRandom
+import java.util.HashMap
+
+open class TestBase {
+    lateinit var convivaConfig: ConvivaConfiguration
+    var convivaAnalytics: ConvivaAnalytics? = null
+    var clientSettingsMock: ClientSettings? = null
+    var clientMock: Client? = null
+    var playerStateManagerMock: PlayerStateManager? = null
+
+    var CONVIVA_SESSION_ID = 1
+    val CUSTOM_EVENT_NAME = "CUSTOM_APPLICATION_EVENT"
+    val CUSTOM_EVENT_ATTRIBUTES: HashMap<String, Any> = mutableMapOf(
+        "key" to "value"
+    ) as HashMap<String, Any>
+    val CUSTOM_ERROR_MESSAGE = "CUSTOM_ERROR_MESSAGE"
+
+    val DEFAULT_DASH_VOD_SOURCE = SourceItem(DASHSource("https://bitmovin-a.akamaihd.net/content/MI201109210084_1/mpds/f08e80da-bf1d-4e3d-8899-f0f6155f6efa.mpd"))
+    val DEFAULT_DASH_VOD_SOURCE_DURATION = 210
+    val DEFAULT_DASH_LIVE_SOURCE = SourceItem(DASHSource("https://livesim.dashif.org/livesim/testpic_2s/Manifest.mpd"))
+    val DEFAULT_DASH_LIVE_SOURCE_DURATION = 0
+    val INVALID_DASH_VOD_SOURCE = SourceItem(DASHSource("https://bitmovin.com"))
+
+    val PLAYER_TYPE = "Bitmovin Player Android"
+    val PLAYER_VERSION = com.bitmovin.player.BuildConfig.VERSION_NAME
+
+    @Before
+    fun setup() {
+        // Create your ConvivaConfiguration object
+        convivaConfig = ConvivaConfiguration()
+        convivaConfig.isDebugLoggingEnabled = true
+    }
+
+    @Before
+    fun tearDown() {
+        convivaAnalytics?.let { it.endSession() }
+
+        // Unmock mocked objects
+        tearDownMocks()
```

Comment:
Following up on my previous comment: This looks like the perfect place to `close` the `activeScenario` 🙂

Reply:
done